spec/observables/dom/ajax-spec.ts

```diff
@@ -1214,45 +1214,70 @@ class MockXMLHttpRequest {
     this.requestHeaders[key] = value;
   }
 
-  protected jsonResponseValue(response: any) {
-    try {
-      this.response = JSON.parse(response.responseText);
-    } catch (err) {
-      throw new Error('unable to JSON.parse: \n' + response.responseText);
-    }
-  }
-
-  protected defaultResponseValue() {
-    if (this.async === false) {
-      this.response = this.responseText;
+  respondWith(
+    response: {
+      status?: number;
+      responseText?: string | undefined;
+      responseType: XMLHttpRequestResponseType;
+      total?: number;
+      loaded?: number;
+    },
+    config?: { uploadProgressTimes?: number; downloadProgressTimes?: number }
```

Comment:
New configuration options for triggering the upload and download progress events (including `loadstart` for each)